Tests/BuildTests/IncrementalBuildTests.swift

```diff
@@ -145,4 +146,28 @@ final class IncrementalBuildTests: XCTestCase {
             XCTAssertNoMatch(log2, .contains("Planning build"))
         }
     }
+    // testing the fix for tracking SDK dependencies to avoid triggering rebuilds when the SDK changes (rdar://115777026)
+    func testSDKTracking() throws {
+#if os(macOS)
+        try XCTSkipIf(!UserToolchain.default.supportsSDKDependentTests(), "skipping because test environment doesn't support this test")
+
+        try fixture(name: "ValidLayouts/SingleModule/Library") { fixturePath in
+            let dummySwiftcPath = SwiftPM.xctestBinaryPath(for: "dummy-swiftc")
+            let swiftCompilerPath = try UserToolchain.default.swiftCompilerPath
+            let environment = [
+                "SWIFT_EXEC": dummySwiftcPath.pathString,
+                "SWIFT_ORIGINAL_PATH": swiftCompilerPath.pathString
+            ]
+            let sdkPathStr = try TSCBasic.Process.checkNonZeroExit(
+                arguments: ["/usr/bin/xcrun", "--sdk", "macosx", "--show-sdk-path"],
+                environment: environment
+            ).spm_chomp()
+
+            let newSdkPathStr = "/tmp/../\(sdkPathStr)"
```

Comment:
```suggestion
            // Generate a new path to make sure the cache is not used and it's rebuilt again with same path.
            let newSdkPathStr = "/tmp/../\(sdkPathStr)"
```

Reply:
Please feel free to change the comment completely if I misunderstood the point of `/tmp/..`, my only suggestion here is that a comment would help when coming back to this test in the future.